lib/watcher-middleware.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var url = require('url');
+var path = require('path');
+
+var setResponseHeaders = require('./utils/response-header');
+var errorHandler = require('./utils/error-handler');
+
+module.exports = function watcherMiddleware(watcher, options) {
+  options = options || {};
+
+  if (!options.hasOwnProperty('autoIndex')) {
+    // set autoIndex to be true if not provided
+    options.autoIndex = true;
+  }
+
+  return function(request, response, next) {
+    watcher.then(function(hash) {
+      var outputPath = path.normalize(hash.directory);
+
+      // set the x-broccoli header containing per request info used by the broccoli-middleware
+      var urlToBeServed = request.url;
+      var urlObj = url.parse(urlToBeServed);
+      var filename = path.join(outputPath, decodeURIComponent(urlObj.pathname));
+      var broccoliInfo = {
+        'url': urlToBeServed,
+        'outputPath': outputPath,
+        'filename': filename,
+      };
+      request.headers['x-broccoli'] = broccoliInfo;
+
+      var updatedFileName = setResponseHeaders(request, response, next, {
+        'url': urlToBeServed,
+        'filename': filename,
+        'outputPath': outputPath,
+        'autoIndex': options.autoIndex
+      });
+
+      if (updatedFileName) {
+        // set the request header file name to be the updated one (could be a null if asset is not found)
+        broccoliInfo.filename = updatedFileName;
+        request.headers['x-broccoli'] = broccoliInfo;
```

Comment:
In what ways can the file name be updated?  This code looks general and the comment makes it sound general, but from what I could tell of `setResponseHeaders` the only change is that `index.html` may be appended.  Did I miss something?